internalsdk/session_model.go

```diff
@@ -114,7 +116,15 @@ func NewSessionModel(mdb minisql.DB, opts *SessionModelOpts) (*SessionModel, err
 	})
 
 	m.baseModel.doInvokeMethod = m.doInvokeMethod
-	return m, m.initSessionModel(context.Background(), opts)
+	go m.initSessionModel(context.Background(), opts)
+	return m, nil
+}
+
+func GetSessionModel() (*SessionModel, error) {
+	if instance == nil {
```

Comment:
I think this is always going to be null.. You are calling `GetSessionModel()` from the Packet Tunnel Provider extension in the iOS client code [here](https://github.com/getlantern/lantern-client/blob/atavism/ios/ios/Tunnel/PacketTunnelProvider.swift#L121), but the session model is setup in the containing app [here](https://github.com/getlantern/lantern-client/blob/atavism/ios/ios/Runner/Lantern/Models/SessionModel.swift#L40)

Reply:
@atavism  yep, That's what's happening, Is there any other way we can do this? Any idea?

Reply:
We need some way to pass data between the provider and the container application.

Reply:
@jigar-f Can we use [sendProviderMessage](https://developer.apple.com/documentation/networkextension/netunnelprovidersession/1406409-sendprovidermessage) for this?

Reply:
Can we do it with [sendProviderMessage](https://developer.apple.com/documentation/networkextension/netunnelprovidersession/1406409-sendprovidermessage) and/or [`handleAppMessage`](https://developer.apple.com/documentation/networkextension/netunnelprovider/1406545-handleappmessage)?

Reply:
Thanks @atavism let me look into it, I think i am close really close to fixing this on that other PR